definitions.h

```diff
@@ -71,6 +76,144 @@ class BoundingBox
         int minY;
 };
 
+class MatrixTransform
+{
+    public:
+        // This is the matrix that will be manipulated into a transform
+        double matrix[4][4];
+
+        // To make operator overloading happy
+        double vector[4][1];
+        
+        // Default Constructor
+        MatrixTransform() {
+            // Declare an Identity Matrix
+            identity();
+        }
+
+        MatrixTransform(double vect[4][1]) {
+            for (int i = 0; i < 4; i++)
+            {
+                this->vector[i][0] = vect[i][0];
+            }
+            identity();
+        }
+    
+        void translate(double x, double y)
```

Comment:
Hard coding these situations work well in the short run but making a function that hard-coded could hurt, maybe making a function that does multiple different S, R, or T in any way that is needed